exasol_integration_test_docker_environment/lib/docker/images/create/docker_image_analyze_task.py

```diff
@@ -122,15 +129,15 @@ def register_required(self):
             tasks = None
         self.dependencies_futures = self.register_dependencies(tasks)
 
-    def keys_are_string(self, task_classes):
+    def keys_are_string(self, task_classes) -> bool:
         return all(isinstance(key, str)
                    for key in task_classes.keys())
 
-    def values_are_subclass_of_baseclass(self, task_classes):
+    def values_are_subclass_of_baseclass(self, task_classes) -> bool:
         return all(issubclass(value, DockerAnalyzeImageTask)
                    for value in task_classes.values())
 
-    def requires_tasks(self) -> Dict[str, Type["DockerAnalyzeImageTask"]]:
+    def requires_tasks(self) -> Optional[Dict[str, Type["DockerAnalyzeImageTask"]]]:
```

Comment:
Not sure about the optional here, might be instead either abstract or returns empty dict, depends on the user of this function, if it accepts none

Reply:
There are tasks which don't override this function.
Actually my initial try with abstract function, but it exploded in the tests.

Reply:
Changed it to the empty dict.